ui/components/app/nft-default-image/toggle-ipfs-modal.js

```diff
@@ -0,0 +1,82 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { useDispatch } from 'react-redux';
+import {
+  Box,
+  ButtonPrimary,
+  Modal,
+  ModalContent,
+  ModalHeader,
+  ModalOverlay,
+  Text,
+} from '../../component-library';
+import { useI18nContext } from '../../../hooks/useI18nContext';
+import { hideIpfsModal, setIpfsGateway } from '../../../store/actions';
+import { IPFS_DEFAULT_GATEWAY_URL } from '../../../../shared/constants/network';
+import {
+  Size,
+  TextColor,
+  TextVariant,
+} from '../../../helpers/constants/design-system';
+
+export const ToggleIpfsModal = ({ onClose }) => {
+  const t = useI18nContext();
+  const dispatch = useDispatch();
+
+  return (
+    <Modal
+      isOpen
+      onClose={() => {
+        onClose();
+      }}
+      className="toggle-ipfs-modal"
+    >
+      <ModalOverlay />
+      <ModalContent>
+        <ModalHeader
+          onClose={() => {
```

Comment:
You can directly use onClose={onClose} here :)

Reply:
Yup, Updated here https://github.com/MetaMask/metamask-extension/pull/20172/commits/dfa2f838aeea7be06f3b23cd391a959251225514